core/src/main/scala/cats/mtl/MonadChronicle.scala

```diff
@@ -0,0 +1,85 @@
+package cats
+package mtl
+
+import cats.data.Ior
+
+trait MonadChronicle[F[_], E] extends Serializable {
+  val monad: Monad[F]
+
+  def dictate(c: E): F[Unit]
+
+  def disclose[A](c: E)(implicit M: Monoid[A]): F[A]
+
+  def confess[A](c: E): F[A]
+
+  def materialize[A](fa: F[A]): F[E Ior A]
+
+  def memento[A](fa: F[A]): F[Either[E, A]]
+
+  def absolve[A](a: => A, fa: F[A]): F[A]
+
+  def condemn[A](fa: F[A]): F[A]
+
+  def retcon[A](cc: E => E, fa: F[A]): F[A]
+
+  def chronicle[A](ior: E Ior A): F[A]
+}
+
+object MonadChronicle {
+  def dictate[F[_], E](e: E)(implicit ev: MonadChronicle[F, E]): F[Unit] = ev.dictate(e)
+
+  def disclose[F[_], A, E](c: E)(implicit ev: MonadChronicle[F, E], m: Monoid[A]): F[A] =
+    ev.disclose(c)
+
+  def confess[F[_], E, A](c: E)(implicit ev: MonadChronicle[F, E]): F[A] = ev.confess(c)
+
+  def materialize[F[_], E, A](fa: F[A])(implicit ev: MonadChronicle[F, E]): F[E Ior A] =
+    ev.materialize(fa)
+
+  def memento[F[_], E, A](fa: F[A])(implicit ev: MonadChronicle[F, E]): F[Either[E, A]] =
+    ev.memento(fa)
+
+  def absolve[F[_], E, A](a: => A, fa: F[A])(implicit ev: MonadChronicle[F, E]): F[A] =
+    ev.absolve(a, fa)
+
+  def condemn[F[_], E, A](fa: F[A])(implicit ev: MonadChronicle[F, E]): F[A] = ev.condemn(fa)
+
+  def retcon[F[_], E, A](cc: E => E, fa: F[A])(implicit ev: MonadChronicle[F, E]): F[A] =
+    ev.retcon(cc, fa)
+
+  def chronicle[F[_], E, A](ior: E Ior A)(implicit ev: MonadChronicle[F, E]): F[A] =
+    ev.chronicle(ior)
+
+  def apply[F[_], E](implicit ev: MonadChronicle[F, E]): MonadChronicle[F, E] =
+    ev
+}
+
+trait DefaultMonadChronicle[F[_], E] extends MonadChronicle[F, E] {
+  override def disclose[A](c: E)(implicit M: Monoid[A]): F[A] = monad.as(dictate(c), M.empty)
+
+  override def memento[A](fa: F[A]): F[Either[E, A]] = monad.map(materialize(fa))(_.toEither)
```

Comment:
I think this should be 
```scala
monad.flatMap(materialize(fa)) {
  case Left(e) => pure(Either.left(e))
  case Right(a) => pure(Either.right(a))
  case Both(e, a) => dictate(e).as(Either.right(a))
}
```

Essentially, a `MonadChronicle` should keep its current record unless it ended with `confess`. See the Haskell docs:

> memento m is an action that executes the action m, returning either its record if it ended with confess, or its final value otherwise, with any record added to the current record.

http://hackage.haskell.org/package/these-0.7.4/docs/Control-Monad-Chronicle.html

Reply:
Hmm, I see. In the `Both` case we'd truncate the left side. Ok, let me fix that